src/nvim/lua/executor.c

```diff
@@ -1229,6 +1231,8 @@ void ex_luado(exarg_T *const eap)
         }
       }
       ml_replace(l, (char_u *)new_line_transformed, false);
+      extmark_splice_cols(curbuf, (int)l-1, 0, (int)old_line_len,
```

Comment:
use `inserted_bytes(l, 0, old_line_len, new_line_len)` when there is an existing `changed_bytes()` call.